WorkloadManagementSystem/DB/JobElasticDB.py

```diff
@@ -0,0 +1,128 @@
+from DIRAC.ResourceStatusSystem.Client.SiteStatus import SiteStatus
+
+#############################################################################
+
+JOB_STATES = ['Submitting', 'Received', 'Checking', 'Staging', 'Waiting', 'Matched',
+              'Running', 'Stalled', 'Done', 'Completed', 'Failed']
+JOB_FINAL_STATES = ['Done', 'Completed', 'Failed']
+
+
+class JobDB(DB):
+
+  def __init__(self):
+    """ Standard Constructor
+    """
+
+    DB.__init__(self, 'jobelasticdb', 'WorkloadManagement/JobElasticDB')
+
+    self.jobAttributeNames = []
+
+    self.siteClient = SiteStatus()
+
+    self.log.info("==================================================")
+
+#############################################################################
+  def getJobParameters(self, jobID, paramList=None):
+    """ Get Job Parameters defined for jobID.
+      Returns a dictionary with the Job Parameters.
+      If parameterList is empty - all the parameters are returned.
+    """
+
+    self.log.debug('JobDB.getParameters: Getting Parameters for job %s' % jobID)
+
+    resultDict = {}
+
+    if paramList:
+      paramNameList = []
+
+      for x in paramList:
+        paramNameList.append(x)
+
+      query = {
+          "query": {
+              "bool": {
+                  "must": [
+                      {"match": {"JobID": jobID}}, {"match": {"Name": ','.join(paramNameList)}}]}},
+          "_source": ["Name", "Value"]}
+
+    else:
+      query = {"query": {"match": {"JobID": jobID}}, "_source": ["Name", "Value"]}
+
+    result = self.query('jobelasticdb*', query)
+
+    if result['OK']:
+      if result['Value']:
+
+        sources = result['Value']['hits']['hits']
+
+        for i in xrange(0, len(sources)):
+
+          name = sources[i]['_source']['Name']
+          value = sources[i]['_source']['Value']
+
+          try:
+            resultDict[name] = value.tostring()
+          except BaseException:
+            resultDict[name] = value
+
+        return S_OK(resultDict)
+
+    else:
+      return S_ERROR('JobDB.getJobParameters: failed to retrieve parameters')
+
+#############################################################################
+  def setJobParameter(self, jobID, key, value):
+    """ Set a parameter specified by name,value pair for the job JobID
+    """
+
+    query = {
+        "query": {
+            "term": {
+                "JobID": jobID}}, "script": {
+            "inline": "ctx._source.Value = params.value; ctx._source.Name = params.name", "params": {
+                "value": value, "name": key}}}
+    indexName = self.generateFullIndexName('jobelasticdb')
+    result = self.exists(indexName)
+
+    if not result:
+      mapping = {
+          "JobParameters": {
+              "properties": {
+                  "JobID": {
+                      "type": "long"}, "Name": {
+                      "type": "text"}, "Value": {
+                      "type": "binary"}}}}
+      result = self.createIndex('jobelasticdb', mapping)
```

Comment:
indexName instead

Reply:
```createIndex()``` uses ```generateFullIndexName()``` first to find the full index name and then creates one if it doesn't exist. ```indexName``` variable has the full name of the index. Using that would create this type of name: ```jobelasticdb-date-date```.

Reply:
yes, You are right!